pkg/registry/deployment.go

```diff
@@ -71,6 +73,16 @@ func GenerateDeployment(cr *registryv1alpha1.DevfileRegistry, scheme *runtime.Sc
 									},
 								},
 							},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "ENABLE_TELEMETRY",
+									Value: strconv.FormatBool(*cr.Spec.Telemetry.Enabled),
```

Comment:
Since `telemetry.enabled` is a bool pointer, won't this cause a panic if it's nil?

We should add a function to defaults.go, `IsTelemetryEnabled` that checks if it's nil (and returns false), otherwise returns the value of the field: